tests/lib/rules/no-this-in-sfc.js

```diff
@@ -217,6 +213,27 @@ ruleTester.run('no-this-in-sfc', rule, {
       return <div onClick={onClick}>{this.props.foo}</div>;
     }`,
     errors: [{message: ERROR_MESSAGE}, {message: ERROR_MESSAGE}]
+  }, {
+    code: `
+    class Foo {
```

Comment:
this isn't a react component, so i wouldn't expect anything in it to be warned on

Reply:
@ljharb 
Similar to current test:

https://github.com/yannickcr/eslint-plugin-react/blob/4c8d8cc0a5dca39554cf97b807822b60782f320f/tests/lib/rules/no-this-in-sfc.js#L136

`Foo` is a class that defines the function `bar` and the function `bar` returns an arrow function that use `this`.

The only difference between this test and the mentioned one is that `bar` is not defined using an arrow function.